File: backend/src/main/java/com/jobboard/service/JobService.java
package com.jobboard.service;

import com.jobboard.dto.request.JobRequest;
import com.jobboard.dto.response.JobResponse;

import java.util.List;

public interface JobService {

    JobResponse createJob(JobRequest jobRequest);

    List<JobResponse> getAllJobs();

    JobResponse getJobById(Long id);

    JobResponse updateJob(Long id, JobRequest jobRequest);

    void deleteJob(Long id);
}

File: frontend/src/services/jobService.js
import axios from "axios";

const API_URL = "https://jobboard-application-globalco-2.onrender.com/api/jobs";

export const getAllJobs = () => {
  return axios.get(API_URL);
};

export const getJobById = (id) => {
  return axios.get(`${API_URL}/${id}`);
};

export const createJob = (job) => {
  return axios.post(API_URL, job);
};

export const updateJob = (id, job) => {
  return axios.put(`${API_URL}/${id}`, job);
};

export const deleteJob = (id) => {
  return axios.delete(`${API_URL}/${id}`);
};